### PR TITLE
Support for all explicit formatting directives in ds9

### PR DESCRIPTION
**Pull request by sushobhana**
See https://github.com/astropy/pyregion/issues/127

### CODE REVIEW

File: regions/io/ds9/read.py [hunk 1]
```diff
@@ -71,7 +71,11 @@ def parse_coordinate(string_rep, unit):
         Parse a single coordinate
         """
         # Any ds9 coordinate representation (sexagesimal or degrees)
-        if 'd' in string_rep or 'h' in string_rep:
+        if string_rep[-1] == 'r':
```
Comment (keflavich) [inline regions/io/ds9/read.py]: comment these lines: # 'r' = radian

File: regions/io/ds9/read.py [hunk 2]
```diff
@@ -71,7 +71,11 @@ def parse_coordinate(string_rep, unit):
...
+            return coordinates.Angle(string_rep[:-1], unit=u.rad)
+        elif string_rep[-1] in ['i', 'p']:
```
Comment (keflavich) [inline regions/io/ds9/read.py]: `# i, p = image, pixel`
(right?)

File: regions/io/ds9/read.py [hunk 3]
```diff
@@ -71,7 +71,11 @@ def parse_coordinate(string_rep, unit):
...
+            return u.Quantity(string_rep[:-1]) - 1
+        elif 'd' in string_rep or 'h' in string_rep:
```
Comment (keflavich) [inline regions/io/ds9/read.py]: then you can move the comment on line 73 here, i.e., h=hours d=degrees

File: regions/io/ds9/read.py [hunk 4]
```diff
@@ -435,8 +441,9 @@ def convert_coordinates(self):
         # We need to copy in order not to burn up the iterators
         elements = [x for x in regex_splitter.split(self.coord_str) if x]
         element_parsers = self.language_spec[self.region_type]
-        for ii, (element, element_parser) in enumerate(
-            zip(elements, element_parsers)):
+        for ii, (element, element_parser) in enumerate(zip(elements,
+                                                           element_parsers)):
+            print(element, element_parser)
```
Comment (keflavich) [inline regions/io/ds9/read.py]: debug statement?

File: regions/io/ds9/read.py (hunk 2 above)
Comment (sushobhana) [reply regions/io/ds9/read.py]: i = image; p=physical according to specs

Comment (sushobhana) [reply regions/io/ds9/read.py]: so yeah!

File: regions/io/ds9/read.py (hunk 4 above)
Comment (sushobhana) [reply regions/io/ds9/read.py]: sorry, forgot to remove this.

**Comment by cdeil** [comment]
@sushobhana @keflavich - Merge?

**Comment by sushobhana** [comment]
this is ready to merge

**Pull request merged**